From cdfa0adaa2dcc558da883d908d58d5ef94c0c0b9 Mon Sep 17 00:00:00 2001
From: "sm1021.shin" <sm1021.shin@samsung.com>
Date: Thu, 7 Nov 2024 16:56:34 +0900
Subject: [PATCH 8/9] Add default maxBufferCount in GraphicBufferProducer

Some modules that inherit and use IGraphicBufferProducer
don't set maxBufferCount. As a result, the maxBufferCount value
that has been queried in libvulkan is 0,
and the creation of swapchain fails.

Therefore, the default value (64) is set to maxBufferCount.

Test: build, Mcts
Bug: 376142095
Bug: 377778846

Change-Id: Ie5c5f747ecd6c15ffae80cc32b9f5a90b1a44df3
Merged-In: Ie5c5f747ecd6c15ffae80cc32b9f5a90b1a44df3
Signed-off-by: sm1021.shin <sm1021.shin@samsung.com>
(cherry picked from commit ec1f230d6db0d5200475a5756951c5f8ee89fab7)
---
 libs/gui/include/gui/IGraphicBufferProducer.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/gui/include/gui/IGraphicBufferProducer.h b/libs/gui/include/gui/IGraphicBufferProducer.h
index 3aac457a09..197e792367 100644
--- a/libs/gui/include/gui/IGraphicBufferProducer.h
+++ b/libs/gui/include/gui/IGraphicBufferProducer.h
@@ -403,7 +403,7 @@ public:
         uint64_t nextFrameNumber{0};
         FrameEventHistoryDelta frameTimestamps;
         bool bufferReplaced{false};
-        int maxBufferCount{0};
+        int maxBufferCount{BufferQueueDefs::NUM_BUFFER_SLOTS};
         status_t result{NO_ERROR};
     };
 
-- 
2.47.1

